### PR TITLE
Fix regex match doc of procfs.PidOf

### LINKED ISSUE #41247: procfs.PidOf description comment is incorrect
The description of `procfs.PidOf` says:

```
// Find process(es) with a specified name (exact match)
```

However the implementation will interpret the name as a regex.  So for example "..." will match any three-letter string. This is not what I understand by "exact match".

https://github.com/kubernetes/kubernetes/blob/e9de1b0221dd8687aba527e682fafc7c33370c09/pkg/util/procfs/procfs_linux.go#L99

### PR DESCRIPTION
**Pull request by feiskyer**
Fixes #41247.

cc @bboreham

### CODE REVIEW

**Comment by brendandburns** [comment]
/lgtm

LGTM from the bulk LGTM tool

**Comment by feiskyer** [comment]
@brendandburns Thanks. Could you help also add approve labels?

**Comment by feiskyer** [comment]
@k8s-bot unit test this

File: pkg/util/procfs/procfs_linux.go [hunk 1]
```diff
@@ -90,7 +90,7 @@ func PKill(name string, sig syscall.Signal) error {
 	return utilerrors.NewAggregate(errList)
 }
 
-// Find process(es) with a specified name (exact match)
+// Find process(es) with a specified name (regexp match)
```
Comment (yujuhong) [inline pkg/util/procfs/procfs_linux.go]: I feel like this is not addressing the issue. The argument is `name` in string, and the function calls `regexp` to make sure it's an exact match.

Comment (feiskyer) [inline pkg/util/procfs/procfs_linux.go]: yes, the name string is treated as a regexp.

**Pull request merged**